### PR TITLE
Furnace count supplemental coil reporting

### LINKED ISSUE #12: Furnace count reporting including supplemental coils
[comstock_sensitivity_reports](https://github.com/NREL/ComStock/blob/main/measures/comstock_sensitivity_reports/measure.rb#L2692-L2749) is treating all gas coils as furnaces (1 gas coil = 1 furnace). Gas coils may be supplemental coils for other kinds of HVAC systems.

The fix is to check whether the coil is the primary heating coil in a unitary system before logging.

### PR DESCRIPTION
**Pull request by LandanT**
Pull request overview
---------------------

This pull request is to update the comstock reporting measure for gas equipment in response to Issues #12. This measure now splits the supplemental and primary gas heating coils into two seperate counts. Updates were made to the measure and the comstock columns definitions .csv

 - Fixes #12

### Pull Request Author

This pull request makes changes to (select all the apply):
 - [ ] Documentation
 - [ ] Infrastructure (includes singularity image, buildstock batch, dependencies, continuous integration tests)
 - [ ] Sampling
 - [ ] Workflow Measures
 - [ ] Upgrade Measures
 - [x] Reporting Measures
 - [x] Postprocessing

Author pull request checklist:

 - [x] Tagged the pull request with the appropriate label (documentation, infrastructure, sampling, workflow measure, upgrade measure, reporting measure, postprocessing) to help categorize changes in the release notes.
 - [x] Register values added to `comstock_column_definitions.csv`
 `test/resource_measure_tests.txt`
 - [ ] All new and existing tests pass the CI
 - [x] Updated measure .xml(s)
### Review Checklist

This will not be exhaustively relevant to every PR.
 - [ ] Perform a code review on GitHub
 - [ ] All related changes have been implemented: data and method additions, changes, tests
 - [ ] If fixing a defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] Reviewed change documentation
 - [ ] Ensured code files contain License reference
 - [ ] Results differences are reasonable
 - [ ] Make sure the newly added measures has been added with tests and indexed properly
 - [ ] CI status: all tests pass

#### ComStock Licensing Language - Add to Beginning of Each Code File
```
# ComStock™, Copyright (c) 2023 Alliance for Sustainable Energy, LLC. All rights reserved.
# See top level LICENSE.txt file for license terms.
```
```

### CODE REVIEW

File: measures/comstock_sensitivity_reports/measure.rb [hunk 1]
```diff
@@ -2719,55 +2719,107 @@ def run(runner, user_arguments)
...
+    # iterate through each model to get all of the gas coils and check if they are supplemental coils for Unitary HVAC Objects and count each seperately.
     model.getCoilHeatingGass.sort.each do |coil|
       # get gas coil capacity
-      capacity_w = 0.0
-      if coil.nominalCapacity.is_initialized
-        capacity_w = coil.nominalCapacity.get
-      elsif coil.autosizedNominalCapacity.is_initialized
-        capacity_w = coil.autosizedNominalCapacity.get
+      supplemental_capacity_w = 0.0
+      if !model.getAirLoopHVACUnitarySystems.empty?
```
Comment (mdahlhausen) [inline measures/comstock_sensitivity_reports/measure.rb]: this selection criteria isn't going to work - 
1) some models may have secondary systems types that are AirLoopHVACUnitarySystems. If the primary system has gas coils, but has AirLoopHVACUnitarySystems, then all gas coils will be labeled as supplemental.
2) Some systems, particularly PSZ-AC RTUs, are modeled as AirLoopHVACUnitarySystems. They have primary gas coils, which will also be logged as supplementary here.

I suggest testing some models with these kinds of variations and seeing if the output variables are logging the capacities into the correct bins.

Comment (mdahlhausen) [reply measures/comstock_sensitivity_reports/measure.rb]: perhaps the more direct method is to call .supplementalCoil and check if the coil name in the loop matches the coil on the systems (in essence, whether the coil is part of a unitary system, and is the supplementary coil on that system.

Comment (LandanT) [reply measures/comstock_sensitivity_reports/measure.rb]: Matt, I have made some updates to the code to use the .supplementalCoil method. We can discuss in person 3/12

**Review by mdahlhausen — APPROVED** [review]
This is working as intended, thanks!
We need to wait to merge until after the release. But once that's done, merge this first, then update the other PRs with it too. You'll need to re-generate the measure .xmls for those Prs.

**Pull request merged**